### PR TITLE
support JSON jamsocket config + address dev cli docker logs race condition

### PR DESCRIPTION
**Pull request by rolyatmax**
More details in the commit messages

### CODE REVIEW

File: cli/src/commands/dev.ts [hunk 1]
```diff
@@ -29,6 +30,32 @@ function isProjectConfig(obj: any): obj is ProjectConfig {
...
+
+  if (hasJson) {
+    const projectConfig = JSON.parse(readFileSync(PROJECT_CONFIG_PATH_JSON, 'utf8'))
+    if (!isProjectConfig(projectConfig)) {
+      throw new Error('Invalid jamsocket.config.json file. Please see https://docs.jamsocket.com/platform/dev-cli for more information.')
+    }
+    return projectConfig
+  }
+
+  // TODO: phase out JS support?
```
Comment (feliciachang) [inline cli/src/commands/dev.ts:48]: why are we phasing out js support?

File: cli/README.md [hunk 2]
```diff
@@ -133,10 +133,11 @@ Starts a local jamsocket dev server. You may configure the dev server with a jam
 
 ```
 USAGE
-  $ jamsocket dev [-d <value>] [-w <value>] [-p <value>] [-i]
+  $ jamsocket dev [-f <value>] [-c <value>] [-w <value>] [-p <value>] [-i]
 
 FLAGS
-  -d, --dockerfile=<value>  Path to the session backend's Dockerfile
+  -c, --context=<value>     Path to the build context for the Dockerfile (defaults to current working directory)
+  -f, --dockerfile=<value>  Path to the session backend's Dockerfile
```
Comment (feliciachang) [inline cli/README.md:140]: why are we switching from -d to -f?

File: cli/src/commands/dev.ts (hunk 1 above)
Comment (rolyatmax) [reply cli/src/commands/dev.ts:48]: It's kind of a pain to get it to work with everybody's JS setup. When you `require()` the config file, it'll throw errors if the package.json has `type: module`. It's also more verbose than JSON (you have to include `module.exports =`, etc. For now we still support both, but I think we should recommend folks use `jamsocket.config.json`

File: cli/README.md (hunk 2 above)
Comment (rolyatmax) [reply cli/README.md:140]: `-f` is used by docker commands to point to a Dockerfile. This keeps backwards compatibility with `-d` though

**Review by feliciachang — APPROVED** [review]

**Pull request merged**